File: app_watcher/src/main/java/com/tongchen/twatcher/gank/ui/adapter/CategoryAdapter.java
package com.tongchen.twatcher.gank.ui.adapter;

import android.text.TextUtils;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseMultiItemQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.tongchen.twatcher.R;
import com.tongchen.twatcher.gank.model.entity.GankResult;
import com.tongchen.twatcher.gank.model.entity.MultipleItem;
import com.tongchen.twatcher.gank.model.http.HttpUrl;
import com.tongchen.twatcher.util.LogUtils;
import com.tongchen.twatcher.util.ScreenUtils;

import java.util.List;

import retrofit2.http.Url;

/**
 * Created by TongChen at 21:46 on 2018/11/1.
 * <p>
 * Description:该文件实现的功能
 */
public class CategoryAdapter extends BaseMultiItemQuickAdapter<MultipleItem, BaseViewHolder> {

    public CategoryAdapter(List<MultipleItem> data) {
        super(data);
        addItemType(MultipleItem.TYPE_TEXT, R.layout.gank_recycle_item_text);
        addItemType(MultipleItem.TYPE_IMG, R.layout.gank_recycle_item_img);
    }

    /*public CategoryAdapter(int layoutResId, @Nullable List<GankResult> data) {
        super(layoutResId, data);
    }*/

    /*@Override
    protected void convert(BaseViewHolder helper, GankResult item) {


    }*/

    @Override
    protected void convert(BaseViewHolder helper, MultipleItem item) {
        switch (item.getItemType()) {
            case MultipleItem.TYPE_TEXT:
                helper.setText(R.id.tv_desc, item.getData().getDesc());
                helper.setText(R.id.tv_date, item.getData().getPublishedAt());
                if (item.getData().getImages() != null && item.getData().getImages().size() > 0) {
                    helper.getView(R.id.iv_preview).setVisibility(View.VISIBLE);
                    Glide.with(mContext).load(item.getData().getImages().get(0)).into((ImageView) helper.getView(R.id.iv_preview));
                } else {
                    helper.getView(R.id.iv_preview).setVisibility(View.GONE);
                }
                break;
            case MultipleItem.TYPE_IMG:
                LogUtils.d("Img", "enter");

                int height = (int) (ScreenUtils.getScreenHeight(mContext) / 3);
//                helper.getView(R.id.iv_preview).getLayoutParams().width = width;
                helper.getView(R.id.iv_preview).getLayoutParams().height = height;

                if (item.getData().getUrl() != null && !TextUtils.isEmpty(item.getData().getUrl())) {
                    helper.getView(R.id.iv_preview).setVisibility(View.VISIBLE);
                    LogUtils.d("Img", item.getData().getUrl() + HttpUrl.IMAGE_WIDTH_SUFFIX + ScreenUtils.getScreenWidth(mContext) / 2);
                    Glide.with(mContext).load(item.getData().getUrl() + HttpUrl.IMAGE_WIDTH_SUFFIX + ScreenUtils.getScreenWidth(mContext) / 2).into((ImageView) helper.getView(R.id.iv_preview));
                } else {
                    helper.getView(R.id.iv_preview).setVisibility(View.GONE);
                }
                break;
        }
    }
}

File: app_watcher/src/main/java/com/tongchen/twatcher/gank/ui/adapter/CategoryAdapter.kt
package com.tongchen.twatcher.gank.ui.adapter

import android.content.Context
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.tongchen.twatcher.R
import com.tongchen.twatcher.base.ui.BaseRecyclerAdapter
import com.tongchen.twatcher.gank.model.entity.GankResult
import com.tongchen.twatcher.gank.model.http.HttpUrl
import com.tongchen.twatcher.gank.ui.fragment.CategoryFragment
import com.tongchen.twatcher.util.ScreenUtils

/**
 * @author TongChen
 * @date 2019/11/20  16:45
 * <p>
 * Desc:
 */
class CategoryAdapter : BaseRecyclerAdapter<GankResult> {

    private var mSpanCount: Int

    constructor(context: Context, spanCount: Int, list: MutableList<GankResult>) : super(context, list) {
        mSpanCount = spanCount

        bindLayoutIds()
    }

    fun bindLayoutIds() {
        mLayouts.put(VIEW_TYPE_HEADER, null)
        if (mSpanCount == CategoryFragment.SINGLE_SPAN_COUNT) {
            mLayouts.put(VIEW_TYPE_BODY, R.layout.gank_recycle_item_text)

        } else {
            mLayouts.put(VIEW_TYPE_BODY, R.layout.gank_recycle_item_img)
        }
        mLayouts.put(VIEW_TYPE_FOOTER, null)
    }

    override fun bindViewHolder(holder: BaseViewHolder, position: Int, itemData: GankResult) {
        when (mSpanCount) {
            CategoryFragment.SINGLE_SPAN_COUNT -> {
                holder.setText(R.id.tv_desc, itemData.desc)
                holder.setText(R.id.tv_date, itemData.publishedAt)
                if (itemData.images != null && itemData.images.size > 0) {
                    holder.getView<ImageView>(R.id.iv_preview).visibility = View.VISIBLE
                    Glide.with(mContext).load(itemData.images[0]).into(holder.getView(R.id.iv_preview))
                } else {
                    holder.getView<ImageView>(R.id.iv_preview).visibility = View.GONE
                }
            }

            CategoryFragment.MULTIPLE_SPAN_COUNT -> {
                val height = (ScreenUtils.getScreenHeight(mContext) / 3).toInt()
                holder.getView<ImageView>(R.id.iv_preview).layoutParams.height = height

                if (itemData.url != null && !TextUtils.isEmpty(itemData.url)) {
                    holder.getView<ImageView>(R.id.iv_preview).visibility = View.VISIBLE
                    Glide.with(mContext).load(itemData.url + HttpUrl.IMAGE_WIDTH_SUFFIX + ScreenUtils.getScreenWidth(mContext) / 2).into(holder.getView(R.id.iv_preview))
                } else {
                    holder.getView<ImageView>(R.id.iv_preview).visibility = View.GONE
                }
            }
        }
    }

}